packages/core/src/xr/XRModule.ts

```diff
@@ -0,0 +1,372 @@
+import { IXRFeatureDescriptor, IXRPlatformFeature } from "@galacean/engine-design";
+import { Engine } from "../Engine";
+import { IXRDevice } from "./IXRDevice";
+import { XRInputManager } from "./input/XRInputManager";
+import { XRFeatureType } from "./feature/XRFeatureType";
+import { XRSessionManager } from "./session/XRSessionManager";
+import { XRSessionState } from "./session/XRSessionState";
+import { XRSessionType } from "./session/XRSessionType";
+import { XRFeatureManager } from "./feature/XRFeatureManager";
+import { XRPlatformFeature } from "./feature/XRPlatformFeature";
+import { Logger } from "../base";
+import { Utils } from "../Utils";
+import { Scene } from "../Scene";
+import { Entity } from "../Entity";
+import { Component } from "../Component";
+
+type TXRFeatureManager = XRFeatureManager<IXRFeatureDescriptor, IXRPlatformFeature>;
+type TXRFeatureManagerConstructor = new (engine: Engine) => TXRFeatureManager;
+type TXRSessionStateChangeListener = (from: XRSessionState, to: XRSessionState) => void;
+type TXRTrackedComponent = new (entity: Entity) => Component;
+
+/**
+ * XRModule is the entry point of the XR system.
+ */
+export class XRModule {
+  // @internal
+  static _featureManagerMap: TXRFeatureManagerConstructor[] = [];
+  // @internal
+  static _componentMap: TXRTrackedComponent[] = [];
+
+  /** Hardware adaptation for XR. */
+  xrDevice: IXRDevice;
+  /** Input manager for XR. */
+  inputManager: XRInputManager;
+  /** Session manager for XR. */
+  sessionManager: XRSessionManager;
+
+  private _engine: Engine;
+  private _scene: Scene;
+  private _origin: Entity;
+  private _features: TXRFeatureManager[] = [];
+  private _sessionState: XRSessionState = XRSessionState.NotInitialized;
+  private _listeners: TXRSessionStateChangeListener[] = [];
+
+  private _mode: XRSessionType;
+  private _requestFeatures: IXRFeatureDescriptor[];
+
+  /**
+   * The current xr scene.
+   */
+  get scene(): Scene {
+    return (this._scene ||= this._engine.sceneManager.activeScene);
+  }
+
+  set scene(value: Scene) {
+    if (this._scene !== value) {
+      this._scene = value;
+      this._origin && value.addRootEntity(this._origin);
+    }
+  }
+
+  /**
+   * The current xr origin.
+   */
+  get origin(): Entity {
+    return (this._origin ||= this.scene?.createRootEntity());
+  }
+
+  set origin(value: Entity) {
+    if (this._origin !== value) {
+      this._origin = value;
+    }
+  }
+
+  /**
+   * The current session mode( AR or VR ).
+   */
+  get mode(): XRSessionType {
+    return this._mode;
+  }
+
+  /**
+   * The requested features.
+   */
+  get requestFeatures(): IXRFeatureDescriptor[] {
```

Comment:
Why you get is a `FeatDescriptor`,but your API is Feat?

Reply:
Done.